src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/vmware/VMWareProviderVirtualMachineUtil.java

```diff
@@ -84,9 +72,110 @@ public VirtualMachine searchVirtualMachineByName(String name, Folder rootFolder)
 
     }
 
-    public Folder searchFolderByName(String name, Folder rootFolder)
-            throws InvalidProperty, RuntimeFault, RemoteException {
+    public Folder searchFolderByName(String name, Folder rootFolder) throws RemoteException {
         return (Folder) new InventoryNavigator(rootFolder).searchManagedEntity("Folder", name);
     }
 
+    public HostSystem searchHostByName(String name, Folder rootFolder) throws RemoteException {
+        return (HostSystem) new InventoryNavigator(rootFolder).searchManagedEntity("HostSystem", name);
+    }
+
+    public Folder searchVMFolderFromVMName(String name, Folder rootFolder) throws RemoteException {
+        ManagedEntity current = searchVirtualMachineByName(name, rootFolder).getParent();
+        while (!(current instanceof Folder)) {
```

Comment:
Could you do it with the search method searchManagedEntities(String[][] typeinfo, boolean recurse) ?